src/components/chip/chip.tsx

```diff
@@ -0,0 +1,279 @@
+ * an action when clicked.
+ *
+ * **Chips:**
+ * Chips are compact elements that represent an input, attribute, or action;
+ * often representing choices, filters, or tags, in a small block
+ * or clearly bundled into a group.
+ * :::
+ *
+ * @private
+ * @exampleComponent limel-example-chip-button
+ * @exampleComponent limel-example-chip-link
+ * @exampleComponent limel-example-chip-icon-colors
+ * @exampleComponent limel-example-chip-badge
+ * @exampleComponent limel-example-chip-filter
+ * @exampleComponent limel-example-chip-removable
+ * @exampleComponent limel-example-chip-aria-role
+ */
+@Component({
+    tag: 'limel-chip',
+    shadow: true,
+    styleUrl: 'chip.scss',
+})
+export class Chip {
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * ID of the chip. Must be unique.
+     */
+    @Prop()
+    id: number | string;
+
+    /**
+     * Label displayed on the chip
+     */
+    @Prop()
+    public text: string;
+
+    /**
+     * Icon of the chip.
+     */
+    @Prop()
+    public icon: string | Icon;
+
+    /**
+     * If supplied, the chip will become a clickable links.
+     */
+    @Prop()
+    link?: Omit<Link, 'text'>;
+
+    /**
+     * The value of the badge, displayed on the chip.
+     */
+    @Prop()
+    badge?: string | number;
+
+    /**
+     * Set to `true` to disable the chip.
+     */
+    @Prop({ reflect: true })
+    public disabled = false;
+
+    /**
+     * Set to `true` to render the chip as a static UI element.
+     * Useful when the parent component has a `readonly` state.
+     */
+    @Prop({ reflect: true })
+    public readonly = false;
+
+    /**
+     * Set to `true` to visualize the chip in a "selected" state.
+     * This is typically used when the chip is used in a chip-set
+     * along with other chips.
+     */
+    @Prop({ reflect: true })
+    public selected = false;
+
+    /**
+     * Set to `true` to render a remove button on the chip.
+     */
+    @Prop({ reflect: true })
+    public removable = false;
+
+    /**
+     * Set to `filter` to render the chip with a distinct style
+     * suitable for visualizing filters.
+     */
+    @Prop({ reflect: true })
+    public type: 'filter';
+
+    /**
+     * Fired when clicking on buttons (not links!)
+     * inside the breadcrumbs.
+     */
+    @Event()
+    public remove: EventEmitter<Chip>;
+
+    @Element()
+    private host: HTMLLimelChipElement;
+
+    private button: HTMLButtonElement;
+    private anchor: HTMLAnchorElement;
+
+    public componentWillLoad() {
+        makeEnterClickable(this.host as HTMLElement);
+    }
+
+    public disconnectedCallback() {
+        this.removeEnterClickable();
+    }
+
+    public render() {
+        if (this.link) {
+            return this.renderAsLink();
+        }
+
+        return this.renderAsButton();
+    }
```

Comment:
Is there a positive reason you want to render a button when there is no link to render, or is it a matter of running into problems using an anchor tag when there is no link to use for the `href`?

Reply:
I don't understand the question :/

Reply:
Do you specifically _want_ to use `<button>` when there is no url, or have you used `<button>` when there isn't a url simply because you had to for some practical reason, and you would be just as happy to use `<a>` in both cases otherwise?

I felt like it was a little complex and annoying to have to handle rendering either an `<a>` or a `<button>`, so I was wondering if we could perhaps make it use an `<a>` even when no url was supplied. But perhaps it's better to use a `<button>` then? 🤔

Reply:
Aha. Yes. for accessibility reasons, I need to render a `<button type="button"`, when the consumer doesn't provide a link.